### ISSUE TITLE
Command auto-completion is very slow on my macOS

### ISSUE THREAD

**Issue by azone** [fish-shell/fish-shell#7575]
The command auto-completion feature is very slow on my macOS. For the detail, please see the gifs below:

fish:
![fish](https://user-images.githubusercontent.com/164407/103057442-7307a500-45da-11eb-8263-4c4dfb4ffdc7.gif)

zsh:
![zsh](https://user-images.githubusercontent.com/164407/103057459-80bd2a80-45da-11eb-9b31-f96968bff6cb.gif)

Here is my environment:

```sh
fish --version                                                                                                                                                 11:24:07
fish, version 3.1.2

 uname -a                                                                                                                                                       11:24:18
Darwin Logans-MacBook-Pro.local 20.2.0 Darwin Kernel Version 20.2.0: Wed Dec  2 20:39:59 PST 2020; root:xnu-7195.60.75~1/RELEASE_X86_64 x86_64

echo $PATH                                                                                                                                                     11:27:11
/usr/local/opt/python@3.9/bin /usr/local/opt/fzf/bin /Users/loganwang/.nodenv/shims /Users/loganwang/.autojump/bin /Users/loganwang/.rbenv/shims /usr/local/bin /usr/bin /bin /usr/sbin /sbin /Library/Apple/usr/bin /Users/loganwang/.cargo/bin/ /Users/loganwang/Downloads/platform-tools/

echo $TERM                                                                                                                                                     11:27:15
xterm-256color
```

**Comment by faho** [comment]
Duplicate of #7565

**Comment by Jane42070** [comment]
The issue is already solved, you can look at #6270 
use this code below, add to your config.fish
`
if test (uname) = Darwin
	set -l darwin_version (uname -r | string split .)
	function __fish_describe_command; end
end
`
